components/linkedin/actions/create-text-post-user/create-text-post-user.mjs

```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+
+export default {
+  key: "linkedin_create-text-post-user",
+  name: "Create a Text Post (User)",
+  description: "Create shares on LinkedIn using text, URLs, and images. [See the docs](https://docs.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/share-on-linkedin#creating-a-share-on-linkedin) for more information",
+  version: "0.0.11",
+  type: "action",
+  props: {
+    name: {
+      type: "string",
+      label: "Text",
+      description: "Text to be posted on LinkedIn timeline",
+    },
+    linkedin: {
+      label: "LinkedIn",
+      description: "LinkedIn Integration",
+      type: "app",
+      app: "linkedin",
+    },
+  },
+  async run({ $ }) {
+    const data = {
+      "author": `urn:li:person:${this.linkedin.$auth.oauth_uid}`,
+      "lifecycleState": "PUBLISHED",
+      "specificContent": {
+        "com.linkedin.ugc.ShareContent": {
+          "shareCommentary": {
+            "text": this.name,
+          },
+          "shareMediaCategory": "NONE",
+        },
+      },
+      "visibility": {
+        "com.linkedin.ugc.MemberNetworkVisibility": "PUBLIC",
+      },
+    };
+    try {
+      await axios.post("https://api.linkedin.com/v2/ugcPosts", data, {
```

Comment:
Move this into the app file, and abstract all API requests using a `_makeRequest` method like so: https://github.com/PipedreamHQ/pipedream/blob/dbc0a14a790a00e8aca8ef5d2fd2cb0aa6c99570/components/github/github.app.js#L197

This keeps things that are common to all LinkedIn API requests — the base URL, the auth headers, the `X-Restli-Protocol-Version`, etc. — within a single method. Then you can rewrite this to pass specific data for this API request. For example, see how the `_makeRequest` method is used in GitHub here: https://github.com/PipedreamHQ/pipedream/blob/dbc0a14a790a00e8aca8ef5d2fd2cb0aa6c99570/components/github/github.app.js#L211

Reply:
Can you review the solution I proposed to reuse the headers, url, method and inject payload? I think it also works very well